### ISSUE TITLE
SIGINT (ctrl+c) during functionapp publish broke syncing triggers

### ISSUE THREAD

**Issue by gabrieljoelc** [Azure/azure-functions-core-tools#3423]
I have been using `func functionapp publish` to deploy to a Linux Consumption Function App:
```zsh
func azure functionapp publish my-linux-consumpmtion-func-app
```
which normally outputs all the triggers when the deployment is complete:
```zsh
Getting site publishing info...
Uploading package...
Uploading 29.16 MB [##############################################################################]
Upload completed successfully.
Deployment completed successfully.
Syncing triggers...
Functions in my-linux-consumpmtion-func-app:
    trigger1 - [httpTrigger]

    trigger2 - [serviceBusTrigger]
```

One of the times I was running it, I Ctrl+C'd during the "Sync triggers..." output:
```zsh
Getting site publishing info...
Uploading package...
Uploading 29.16 MB [##############################################################################]
Upload completed successfully.
Deployment completed successfully.
Syncing triggers...
^C # <======================= RIGHT HERE
```

Now when I run the same command, I doesn't output the triggers and show this:
```zsh
Getting site publishing info...
Uploading package...
Uploading 29.16 MB [##############################################################################]
Upload completed successfully.
Deployment completed successfully.
Syncing triggers...
Functions in my-linux-consumpmtion-func-app:
```

### _The triggers fail to sync and the portal shows I have no triggers._

Things I've tried:
- Deleting and recreating the function app resource
- Uninstalling and reinstalling Azure Functions Core Tool CLI (via homebrew)
- Re-logging in with `az login`
- Restarting my computer
- Uninstalling and reinstalling az CLI and removing `~/.azure` directory

## Update 1

I checked the `WEBSITE_RUN_FROM_PACKAGE` setting in the function app and it does have the correct URL of the latest release. I curl the URL, downloaded the file as a zip, and verified that it contains the built application code. 🤷 

Is it something related to the SIGINT breaking the CLI so it can't bump the `azure-webjobs-hosts/synctriggers/my-linux-consumption-func-app/last` blob anymore or something?

**Comment by danrimbault** [comment]
@gabrieljoelc I don't know specifically about crtl c'ing when the sync is underway. But I think there are issues with azure-functions-core-tools. Our application was deploying and syncing fine from local scripts on windows machines. But on the pipelines (linux image) the tool would report successful sync but then there would be no functions in the app.
Upgrading azure-functions-core-tools to v4 (which seems to introduce dependency on dotnet sdk) seems to have solved the problem. 
TLDR be suspicious of azure-functions-core-tools v3 output when publishing
